tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -1608,12 +1617,19 @@ bool SKDocConsumer::handleDiagnostic(const DiagnosticEntryInfo &Info) {
 // ReportCursorInfo
 //===----------------------------------------------------------------------===//
 
-static void reportCursorInfo(const CursorInfoData &Info, ResponseReceiver Rec) {
-
+static void reportCursorInfo(const CursorInfoData &Info, ResponseReceiver Rec,
+                             StringRef Error) {
   if (Info.IsCancelled)
     return Rec(createErrorRequestCancelled());
+  if (!Error.empty())
+    return Rec(createErrorRequestFailed(Error.str().c_str()));
```

Comment:
I understand these APIs are pre-existing, but there's an opportunity to improve them.. Looking at the implementation of `createErrorRequestFailed` it doesn't seem useful to have it accepting a `const char *`, we could have it accept a `StringRef` instead to simplify the callers and avoid unnecessary `std::string` conversions.

Reply:
Modified the API